ASSETSELECTION.md

```diff
@@ -0,0 +1,76 @@
+## expo-updates asset selection demo
+
+_Note:_ This app uses new features that are still under development, and not yet part of any Expo SDK release. It is provided to allow these features to be previewed, and is for demonstration purposes only.
+
+### Introduction to asset selection
+
+In SDK 49 and earlier, all assets resolved in the Metro bundler are included in every update, and are uploaded to the update server. The new experimental asset selection feature in SDK 50 allows the developer to specify that only certain assets should be included in updates. This has the potential to greatly reduce the number of assets that need to be uploaded to and downloaded from the updates server.
+
+To use this feature, a new property `extra.updates.assetPatternsToBeBundled` should be included in your Expo config. It should define one or more file matching patterns (regular expressions). For example, suppose your `app.json` file has the patterns defined this way:
+
+```json
+    "extra": {
+      "updates": {
+        "assetPatternsToBeBundled": [
+          "app/images/**/*.png"
+        ]
+      }
+    }
+```
+
+In this case, all `.png` file in all subdirectories of `app/images` would be included in updates.
+
+Assets that do not match any of the file patterns will still resolve in the Metro bundler, but will not be uploaded to the updates server. If making use of this feature, a developer needs to be certain that assets not included in updates are built into the native build of the app.
+
+If this new property is not in your Expo config, all assets resolved by the bundler will be included in updates (SDK 49 and earlier behavior).
+
+### Quick start
+
+This demo makes use of a component that displays image assets. To initialize this component with the default images from `app/embeddedImages`, execute
+
+```bash
+yarn select-images
+```
+
+Then follow the quick start instructions in the [README](./README.md) document to do an initial build of the app:
+
+```bash
+yarn
+eas init
+eas update:configure
+# Build and run iOS
+npx expo run:ios --configuration Release
+# Or build and run Android
+npx expo run:android --variant release
+```
+
+### Embedded assets
+
+The build will include two image files, both in the `app/embeddedImages` folder. These assets will be available to the initial build, and to all subsequent updates. When running the initial build, you will see these two images in the "Image Assets" card on the home screen.
+
+![assets-01-embedded-images](./media/assets-01-embedded-images.jpg)
+
+### Assets included in updates
```

Comment:
Suppose there was an image in **embeddedImages**, and you changed it very slightly after your build, and then published an update. Would that cause that image to now show as blank in the app, since the hash would no longer match? Not sure if it should be emphasized here, in the docs, or both, but seems like something to note - that just because the image matches a path that was included in the embedded bundle doesn't mean an update will fall back to that (if that is the case)

Reply:
You are right. If you modify an existing image that is part of the embedded assets, it would no longer show unless it is moved to a location that is included in updates. We should make this scenario part of this demo.